### PR TITLE
Make acceptable lost particles accessible

### PR DESCRIPTION
**Pull request by stevendargaville**
This is a PR to address #1471, where the hard-coded const expressions for max_lost_particles and relative max_lost_particles are now added to the settings class and hence accessible to the C++/Python/XML layers. Also added to the regression/unit tests to check these variables can be read/written correctly. 

The defaults for these variables have been retained so there should be no changes to any existing tests, etc.

### CODE REVIEW

File: include/openmc/settings.h [hunk 1]
```diff
@@ -63,10 +63,12 @@ extern std::string path_source;
 extern std::string path_sourcepoint;      //!< path to a source file
 extern "C" std::string path_statepoint;   //!< path to a statepoint file
 
-extern "C" int32_t n_batches;      //!< number of (inactive+active) batches
-extern "C" int32_t n_inactive;     //!< number of inactive batches
-extern "C" int32_t gen_per_batch;  //!< number of generations per batch
-extern "C" int64_t n_particles;    //!< number of particles per generation
+extern "C" int32_t n_batches;                //!< number of (inactive+active) batches
+extern "C" int32_t n_inactive;               //!< number of inactive batches
+extern "C" int32_t n_max_lost_particles;     //!< maximum number of lost particles
```
Comment (paulromano) [inline include/openmc/settings.h]: I would suggest removing the `n` to be consistent with the Python side:
```suggestion
extern "C" int32_t max_lost_particles;       //!< maximum number of lost particles
```

File: openmc/lib/settings.py [hunk 2]
```diff
@@ -22,6 +22,8 @@ class _Settings(object):
     entropy_on = _DLLGlobal(c_bool, 'entropy_on')
     generations_per_batch = _DLLGlobal(c_int32, 'gen_per_batch')
     inactive = _DLLGlobal(c_int32, 'n_inactive')
+    max_lost_particles = _DLLGlobal(c_int32, 'n_max_lost_particles')
```
Comment (paulromano) [inline openmc/lib/settings.py]:
```suggestion
    max_lost_particles = _DLLGlobal(c_int32, 'max_lost_particles')
```

File: src/state_point.cpp [hunk 3]
```diff
@@ -93,6 +93,8 @@ openmc_statepoint_write(const char* filename, bool* write_source)
     write_attribute(file_id, "photon_transport", settings::photon_transport);
     write_dataset(file_id, "n_particles", settings::n_particles);
     write_dataset(file_id, "n_batches", settings::n_batches);
+    write_dataset(file_id, "n_max_lost_particles", settings::n_max_lost_particles);
+    write_dataset(file_id, "relative_max_lost_particles", settings::relative_max_lost_particles);
```
Comment (paulromano) [inline src/state_point.cpp]: I don't see a compelling reason to put these in the statepoint file. There are many other settings that we don't store in the statepoint. The purpose is really to save information from a simulation as opposed to what settings were in use.

File: tests/regression_tests/photon_source/test.py [hunk 4]
```diff
@@ -52,6 +54,8 @@ def _get_results(self):
             outstr += 'tally {}:\n'.format(t.id)
             outstr += 'sum = {:12.6E}\n'.format(t.sum[0, 0, 0])
             outstr += 'sum_sq = {:12.6E}\n'.format(t.sum_sq[0, 0, 0])
+            outstr += 'max_lost = {:12.6E}\n'.format(sp.n_max_lost_particles)
+            outstr += 'rel_max_lost = {:12.6E}\n'.format(sp.relative_max_lost_particles)
```
Comment (paulromano) [inline tests/regression_tests/photon_source/test.py]: Along the same lines, if these are removed from the statepoint, they shouldn't be in the test either. I think it's sufficient to have the unit test that confirms the proper values were put into place.

File: src/state_point.cpp (hunk 3 above)
Comment (stevendargaville) [reply src/state_point.cpp]: Ah good to know, I was a little uncertain if they should be output by the statepoint.

File: include/openmc/settings.h (hunk 1 above)
Comment (stevendargaville) [reply include/openmc/settings.h]: No problem, will do

**Comment by stevendargaville** [comment]
Thanks for the suggestions, I've made the requested changes and looks like all the tests are green.

**Comment by paulromano** [comment]
Thanks @stevendargaville. Congrats on your first PR! :tada:

**Pull request merged**